### DISCUSSION TITLE
크로스 브라우징에 대해 설명해주세요 + 크로스브라우징을 고려해 작업한 스스로의 경험이 있는 경우 설명해주세요.

### DISCUSSION THREAD

**Discussion by bona373737** [wanted-9th-3team/frontend-study#37] [프론트 기술 질문 공간]
크로스브라우징

**Comment by michoball** [comment]
크롬, IE, 파이어폭스, 엣지, 오페라 등 다양한 웹 브라우저를 사용하는 유저에게 동일한 UI/UX를 제공하기 위해서  W3C에서 정한 웹 규격에 맞게 html, css, javascript를 작성하는 것을 말합니다. 
브라우저마다 사용하는 js 엔진이 다르고 업데이트 되는 내용이 다르기 때문에 크롬에서의  UI/UX가 파이어폭스나 IE에서는 이상하게 보일 수 있습니다. 그래서 크로스 브라우징을 고려해 작업을 해야합니다. 

예전 팀 프로젝트를 진행할 당시 display : flex로 정의한 div 안의 자식 노드들 간 거리를 gap으로 정한 적이 있었습니다. 그 당시 caniuse 웹사이트 상에 따르면 대부분의 브라우져에서는 잘 보여지나 IE에서는 gap이라는 속성을 인식하지 못한다는 피드백이 있어서 IE 사용자가 아직 많은 우리나라의 특성을 고려해 gap을 사용하지 않고 margin-left나 margin-right를 사용하여 거리를 조정한 적이 있습니다.

**Comment by skyhanull** [comment]
크로스브라우징이란 웹페이지의 상호 호환성을 의미합니다. 이는 크롬, IE, 파이어폭스, 엣지, 오페라  브라우저마다 렌더링 엔진이 다르기 때문입니다. 특히 IE는 호환이 별로 좋지 않다고 합니다
단 어느 한 브라우저에 최적화해서 만드는 것이 아닌 공통되게 만드는 것이 좋으나 점유율이 많은 쪽위주로 하면 좋다고 합니다.
실제로는 잘모르겠으나 과제에서 크로스 브라우징 검사가 조건에 있어서 lambda test를 이용해본 경험이 있습니다.

**Comment by WooGie911** [comment]
크로스 브라우징 () 은 웹 페이지의 상호 호환성을 의미합니다. 

쉽게 말하면 서로 다른 브라우저들끼리 서로 호환이 가능하도록 하는 것을 의미합니다.

어느 한쪽에 최적화되지 않고, 공통 요소를 사용하여 동등한 수준의 정보, 기능 접근에 초점을 두고 개발하는 것이 좋습니다.

하지만 점유율이 높은 쪽을 위주로 개발하는 것이 현실이기에 사람들이 주로 사용하는 브라우저가 어떤게 있는지 알아본 후 그 브라우저를 타겟으로 잡아 대응하는 것이 좋다고 생각합니다.

또한 '모질라 MDN', 'Can I Use...' 등을 통해 각 브라우저의 호환성을 검토하는 과정도 필요하다 생각합니다.

**Comment by javascriptbiggosoo** [comment]
해당 이슈가 있다느 점은 인지하고 있지만 최근 safari를 제외한 브라우저들이 웹표준을 잘 지키고 있어서 직접 문제를 경험해본 적은 없습니다.
